### PR TITLE
Feature/fix auth and session state

### PR DESCRIPTION
**Pull request by simoexpo**
During some test I discovered that I had some problem on using the lib with some rtsp url that needs an authentication. Plus I found that in my test the session_id field was not properly propagated.

This PR should address the above problems:
- Properly update the auth details in the session state
- Properly include the session_id header in the requests

Please let me know if I missed something or if they are really some bug that should be fixed

Thanks :)

### CODE REVIEW

File: test/membrane_rtsp/workflow_integration_test.exs [hunk 1]
```diff
@@ -41,6 +41,7 @@ defmodule Membrane.RTSP.WorkflowIntegrationTest do
   end
 
   def resolver(request) do
+    IO.inspect(request)
```
Comment (mat-hek) [inline test/membrane_rtsp/workflow_integration_test.exs]: please remove

**Review by mat-hek — COMMENTED** [review]
Hi @simoexpo, It seems they were bugs, thanks for the fixes!

File: test/membrane_rtsp/workflow_integration_test.exs (hunk 1 above)
Comment (simoexpo) [reply test/membrane_rtsp/workflow_integration_test.exs]: Sorry I forgot it. Removed :)

**Review by mat-hek — APPROVED** [review]

**Comment by mat-hek** [comment]
released as v0.2.1

**Pull request merged**